S21/indraneelL/refactorme.py

```diff
@@ -1,84 +1,86 @@
+        return temp.lower()
+
+class CheckTurkAndAzerWord(BaseClass):
+
+    def process_word(self):
+        temp = self.word
+        temp = temp.replace(self.CAPITAL_I,self.DOTLESS_i)
+        return temp.lower()
+
+class CheckGreekWord(BaseClass):
+
+    def process_word(self):
+        temp = self.word
+        if temp[-1]== self.CAPITAL_SIGMA:
+            temp = temp[:-1]+ self.CAPITAL_SIGMA
+        return temp.lower()
+
 class Word:
 
-  def __init__(self, word, bcpCode, std=False):
-    self._w = word
-    self._l = bcpCode
-    self._finalSigma = False
-    self._standardIrishSpelling = std
-    # OLD EXPERIMENTAL CODE for dealing with vowel harmony
-    # self._numVowels = 0
-    # for c in word:
-    #   if c in 'aeiouAEIOU':
-    #   self._numVowels += 1
-
-  def setWord(self, w):
-    self._w = w
-
-  def toLower(self):
-    language = self._l
-    if '-' in self._l:
-      i = self._l.find('-')
-      language = self._l[0:i]
-    if len(language)<2 or len(language)>3:
-      print("Invalid BCP-47 code")
-      return ''
-    temp = self._w
-    if language=='zh':
-      return temp
-    elif language=='ja':
-      return temp
-    elif language=='ga':
-      if len(self._w)>1:
-        if (self._w[0]=='t' or self._w[0]=='n') and unicodedata.normalize('NFC', self._w)[1] in 'AEIOU\u00c1\u00c9\u00cd\u00d3\u00da':
-          temp = self._w[0]+'-'+temp[1:]
-      return temp.lower()
-    elif language=='tr':
-      temp = self._w
-      temp = temp.replace('\u0049','\u0131')
-      return temp.lower()
-    elif language=='az':
-      temp = self._w
-      temp = temp.replace('\u0049','\u0131')
-      return temp.lower()
-    elif language=='th':
-      return temp
-    elif language=='el':
-      if temp[-1]=='\u03a3':
-        self._finalSigma = True
-        temp = temp[:-1]+'\u03c2'
-      return temp.lower()
-    elif False and language=='gd':
-      # specification doesn't ask for this language to be treated differently
-      # so this will never be called
-      if len(self._w)>1:
-        if (self._w[0]=='t' or self._w[0]=='n') and self._w[1] in 'AEIOU\u00c1\u00c9\u00cd\u00d3\u00da':
-          temp = self._w[0]+'-'+temp[1:]
-      return temp.lower()
-    else:
-      return temp.lower()
-
-  def isLenited(self):
-    language = self._l
-    if '-' in self._l:
-      i = self._l.find('-')
-      language = self._l[0:i]
-    if language == 'ga' or language == 'gd':
-      if len(self._w) < 2:
-        return False
-      else:
-        return self._w[0].lower() in 'bcdfgmpst' and self._w[1].lower()=='h'
-    else:
-      raise NotImplementedError('Method only available for Irish and Scottish Gaelic')
+    def __init__(self, word, lang):
+        self._w = word
+        self._l = lang
+
+    def checkForLangCode(self):
+        language = self._l
+        if '-' in self._l:
+            i = self._l.find('-')
+            language = self._l[0:i]
+        if language == "az" or language == "tr":
```

Comment:
I believe we need to have language tags be "ga" and "gd" instead of "az" "tr" here.

Reply:
No, as the functionality for "Turkish" and "Azer" is the same I just replaced the language variable with a common one which is "at", which performs the same operation for both languages. Removed some of the languages as they are not in tests.tsv. I'm not sure if we need to refactor based on tests.tsv or consider all the languages which are already in the professor's code.